### PR TITLE
Support typehints for various of ArrayType.

### LINKED ISSUE #1133: Support for Array type hints in APIs that take Python native functions (e.g., DataFrame.apply)
```python
import databricks.koalas as ks

def tokenizeDF(col1) -> ks.Series[np.array(... ???)]:
    pass
```

### PR DESCRIPTION
**Pull request by itholic**
Until now, only arrays of string type could be used as type hints with np.array.
This PR proposes to support many other child types, too.

This should resolve #1133

### CODE REVIEW

File: databricks/koalas/typedef/typehints.py [hunk 1]
```diff
@@ -22,6 +22,11 @@
 import decimal
 from inspect import getfullargspec, isclass
 
+try:
+    from typing import _GenericAlias  # type: ignore
```
Comment (itholic) [inline databricks/koalas/typedef/typehints.py:26]: FYI: `GenericMeta` is renamed to `_GenericAlias` in Python3.7.

File: databricks/koalas/typedef/typehints.py [hunk 2]
```diff
@@ -104,8 +109,9 @@ def as_spark_type(tpe) -> types.DataType:
     # TODO: Add "boolean" and "string" types.
     # ArrayType
     if tpe in (np.ndarray,):
-        # TODO: support other child types
         return types.ArrayType(types.StringType())
+    elif isinstance(tpe, _GenericAlias) and (tpe.__origin__ in (list, typing.List)):
```
Comment (ueshin) [inline databricks/koalas/typedef/typehints.py:113]: Shall we use `hasattr(tpe, "__origin__")` to use the same way as in `infer_return_type`?

Comment (ueshin) [reply databricks/koalas/typedef/typehints.py:113]: Btw, I don't think `list` can be a generic type?
In that case, should we handle `list` as the same as `np.ndarray`? WDYT @HyukjinKwon ?

**Review by ueshin — APPROVED** [review]
LGTM.

**Comment by ueshin** [comment]
Thanks! merging.

**Comment by itholic** [comment]
Thanks for the review, @ueshin

**Pull request merged**